include/openmc/particle_data.h

```diff
@@ -264,6 +264,8 @@ class ParticleData {
 
   // Other physical data
   double wgt_ {1.0};       //!< particle weight
+  double wgt_cutoff_;      //!< particle weight cutoff
```

Comment:
I think we can probably get away with only adding the source weight to particles rather than both a cutoff and survival weight. perhaps something along the lines of `double wgt0_ {1.0};` Then the other computations can happen on the fly I think?